MusicTaste/MusicTaste/View Controllers/MatchmakingViewController.m

```diff
@@ -16,58 +16,66 @@ @interface MatchmakingViewController ()<UITableViewDataSource, UITableViewDelega
     self.MatchTableView.dataSource = self;
     self.MatchTableView.delegate = self;
     self.refreshControl = [[UIRefreshControl alloc] init];
     [self.MatchTableView insertSubview:self.refreshControl atIndex:0];
     [self.MatchTableView addSubview:self.refreshControl];
     self.MatchTableView.rowHeight = 540;
+    //Checks if Notification has been updated
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(updateView:)
+                                                 name:@"TestNotification"
+                                               object:nil];
     MatchCell *cell = [_MatchTableView dequeueReusableCellWithIdentifier:@"MatchCell"];
     [self.view bringSubviewToFront:cell.matchName];
+}
+
+- (void)updateView:(BOOL)animated{
+    //Function to update matches once the notifcation is returned
+    self.author  = [SpotifyAPIManager shared].author;
     [self getMatchesDictionary];
     [[MatchingAlgorithm shared] lookForMatches];
+    [self getMatchesDictionary];
 }
-    
+
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
     return self.matches.count;
 }
 
 - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
     MatchCell *cell = [tableView dequeueReusableCellWithIdentifier:@"MatchCell"];
     PFObject *match = self.matches[indexPath.row];
+    //returns Genres, bio, name, and user's Artists
     cell.matchName.text = [match valueForKey:@"username"];
+    cell.matchBio.text = [match valueForKey:@"bio"];
     cell.matchArtists.text = [[[match valueForKey:@"images"] objectAtIndex:0] objectAtIndex:0];
     cell.matchGenres.text = [[match valueForKey:@"genres"] objectAtIndex:0];
     if ([match valueForKey:@"userimage"] != nil){
         NSString *URLString = [match valueForKey:@"userimage"];
         NSString *stringWithoutNormal = [URLString stringByReplacingOccurrencesOfString:@"_normal" withString:@""];
         NSURL *urlNew = [NSURL URLWithString:stringWithoutNormal];
         [cell.matchImage setImageWithURL: urlNew];
+        cell.matchImage.layer.cornerRadius = cell.matchImage.frame.size.height/10;
     }
     return cell;
 }
 
 -(void)getMatchesDictionary{
-    //goes through users to get all the matches user's ID
-    PFUser *current = [PFUser currentUser];
     PFQuery *music = [PFQuery queryWithClassName:@"Music"];
-    NSArray *musicUsers = [music findObjects];
-    for (PFObject *user in musicUsers){
-        if ([[user valueForKey:@"userId"] isEqual:current.objectId]){
-            NSMutableArray *allMatchesArray = [user valueForKey:@"matches"];
-            NSMutableDictionary *matchesDict = [allMatchesArray objectAtIndex:0];
-            self.matchesObjectIds = [matchesDict keysSortedByValueUsingSelector:@selector(compare:)];
-        }
-    }
+    NSMutableArray *allMatchesArray = [self.author valueForKey:@"matches"];
```

Comment:
Makes it much simplier once I have the user and don't have to create a callback

Reply:
Definitely looks like a good simplification of code. Which callback though? I don't see one here.

Reply:
I am referring to earlier code in MatchmakingViewController so I wouldn't have to compare to see the users were matching with the user id.